### ISSUE TITLE
ManagedMqttClient is not working in a IHostedService environment on MacOS

### ISSUE THREAD

**Issue by dannyyy** [dotnet/MQTTnet#1168]
### Describe the bug
Following the `ManagedMqttClient` example on MacOS does absolutely nothing.
Client has `IsStarted` set to tue but `IsConnected` stays at false. There is no error, no exception nothing.
I can also set the hostname to `thisis.carp` and there is still no error. Using TLS or not does not change anything.

If I try to publish or ping immediately after `await client.StartAsync(options)` I receive either a `NullReferenceException` or a `TaskCancelledException`. In rare cases I get a `IndexOutOfBoundException`.

After an exception has been thrown, there is no way to stop debugging or exit the IDE. On MacOS only a `kill -9` can terminate the process. Another sign of suspicious behavior. 

And there is absolutely no logging output, not a single line. I can only guess, that this has to do with a IHostedService. But then the library has a really bad and unexpected design. Even not having the ability to use DI is almost a no-go.

### Which project is your bug related to?

- ManagedClient

### To Reproduce
*Startup.cs*
```
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddControllers();
            services.AddSwaggerGen(c =>
            {
                c.SwaggerDoc("v1", new OpenApiInfo {Title = "Inkbird.Backend", Version = "v1"});
            });

            services.AddHostedService<MqttHostedService>();
        }
```

*MqttHostedService*
```
using System;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.Extensions.Hosting;
using Microsoft.Extensions.Logging;
using MQTTnet;
using MQTTnet.Client.Options;
using MQTTnet.Extensions.ManagedClient;

namespace MyProject.Backend.Backend
{
    public class MqttHostedService : IHostedService
    {
        private readonly ILogger<MqttHostedService> _logger;
        private IManagedMqttClient _mqttClient;

        public MqttHostedService(ILogger<MqttHostedService> logger)
        {
            _logger = logger;
        }
        
        public async Task StartAsync(CancellationToken cancellationToken)
        {
            _logger.LogInformation("Start MQTT background service");

            var clientOptions = new MqttClientOptionsBuilder()
                .WithClientId("My-Backend-Client")
                .WithCredentials("username", "password")
                .WithTcpServer("mqtt.mydomain.com", 8883)
                .WithTls()
                .Build();
            
            var managedOptions = new ManagedMqttClientOptionsBuilder()
                .WithAutoReconnectDelay(TimeSpan.FromSeconds(5))
                .WithClientOptions(clientOptions)
                .Build();
            
            _mqttClient = new MqttFactory().CreateManagedMqttClient();

            await _mqttClient.StartAsync(managedOptions);
            await _mqttClient.PublishAsync("abc", "ahd");
        }

        public async Task StopAsync(CancellationToken cancellationToken)
        {
            _logger.LogInformation("Stop MQTT background service");

            if (_mqttClient != null && _mqttClient.IsStarted)
                await _mqttClient.StopAsync();
        }
    }
}
```
### Expected behavior
Nothing less than an established connection or clear error messages.

### Additional context / logging
There is no logging output, no library exception, nothing. The given exceptions are thrown by .NET Task/Await/Async code.

**Comment by nnilsson123** [comment]
I'm experiencing the same behavior with ManagedClient - .StartAsync does not cause the managed client to be in a Connected state. No exceptions, nothing. Very confusing...

**Comment by xeonfusion** [comment]
Similar issues with Managed client and WebSockets, fails (or takes a long time) to connect to a test Mosquitto broker, with .NET 5.0 on Windows 10, and fails on Mac OS X. It does work sometimes if .NET standard framework or Mono is used on Mac OS X or Linux. I suspect platform specific code issues with the ManagedClient code (messages getting enqueued without actually being published, Task or threading issues?). @chkr1011 Is there any recommended way (examples or unit tests) to use the ManagedClient with WebSockets on .NET 5.0?

**Comment by dannyyy** [comment]
Very disappointing. After half a year still no official feedback. I gave another try with the new 4.0-preview but still buggy as hell. Even this library seems to be periodically maintained, a serious mindset is not behind.

My suggestion to the maintainer, just get rid of the `ManagedMqttClient` and unlist any NuGet package related to this.

**Comment by chkr1011** [comment]
The managed client is indeed not the primary focus in this project. I often recommend using the regular client because the managed one only covers the most simple use cases. When it comes to more advanced features the managed client will never be able to match all special cases (send custom user properties, custom retry behavior with backup servers etc.). 

I will have a look regarding this issues when version 4.0.0 of this library is out.

**Comment by dagophil** [comment]
I dont think that there is a general problem with the managed client in hosted services. Maybe it is a specific problem on MacOS or when using WebScokets?

In the past year, I used the managed client in a hosted service through different versions of MQTTnet (starting with 3.0.13). During that time, I utilized it in several applications that target Windows or Linux, use .NET 5 or .NET 6, and run on desktop or in a docker environment. The code of the hosted service is similar to the above and it worked without any issues.
